src/js/common/schemaform/SchemaForm.jsx

```diff
@@ -57,6 +58,25 @@ class SchemaForm extends React.Component {
     }
   }
 
+  /*
+   * If we're in review mode, we can short circuit updating
```

Comment:
The review page is slow mainly if you open several chapters and then edit a page, because it's re-rendering all of the visible pages on every character you type. This cuts down on that.